v3io/common/helpers.py

```diff
@@ -12,16 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-def url_join(base, *parts):
-    result = base
 
-    if result[0] != "/":
-        result = "/" + base
 
-    for part in parts:
-        if part[0] != "/":
+def url_join(*parts):
```

Comment:
I think this might be clearer:
```python
def url_join(*parts):
    slash_suffix = False
    result = ""
    for part_index, part in enumerate(parts):
        if part == "":
            continue
        if slash_suffix:
            # if slash suffix existed in prev trim slash prefix from this part
            result += part.lstrip("/")
        else:
            # add slash prefix before part if:
            # 1. slash suffix did not exist in prev part and
            # 2. slash prefix does not exist in this part and
            # 3. part is not the first
            if part[0] != "/" and part_index != 0:
                result += "/"
            result += part

        slash_suffix = part[-1] == "/"
    return result
```